sqlserver/tests/test_statements.py

```diff
@@ -452,9 +452,7 @@ def test_statement_metadata(
 ):
     check = SQLServer(CHECK_NAME, {}, [dbm_instance])
 
-    query = '''
-    -- Test comment
-    select * from sys.databases'''
```

Comment:
Can we perhaps keep the original query with the comment after adding the preceding `dbcc freeproccache` call?

Reply:
No, in that case the test failed.

Before this PR, this query is not important.
For example, In the case of different comments (https://github.com/it-ito/integrations-core/commit/63d4ab8e291dc68bd90d898bdf139983df0b96fa)  or different table (https://github.com/it-ito/integrations-core/commit/228a9dd3897c99822d84c67fe125af66c6cb9e2d) , test case succeed before. 

Perhaps there was a `select * from sys.databases` running somewhere that was being used.

Reply:
I see, thanks. The query is being used by Agent for auto-discovery and it's possible that it interferes with the test case.

But do we then need `dbcc freeproccache` call at all?

Does it work if we remove `dbcc freeproccache` and replace `sys.databases` with `sys.sysusers`, as the latter isn't queried by the agent? That would be the perfect solution because it would both fix the original problem and preserve the test case with the comment. If that would be too much to ask, we can go further with your changes and later deal with the test case problem internally. 

Thanks for your help!